### ISSUE TITLE
Новые ссылки фейсбука нормально не парсятся

### ISSUE THREAD

**Issue by DJ-ZX** [rekcuFniarB/forum-theprodigy-ru#147]
https://www.facebook.com/rgorin.zx/posts/3303345403026764?__cft__[0]=AZUIklBxaMP74zvjQ0rBVgic8lY3bjcJkI1CCszL-pPJhmGMo1BTXTO8SyyeXvnfuVtKONqdsZTOiVP-LUmZgCixOO883aSZcN0RuV8jnSE95oKyXMHtTZjBkHRGP4_WVq0&__tn__=%2CO%2CP-R

превращается в 
![image](https://user-images.githubusercontent.com/13614247/77257619-2d4e3280-6c7e-11ea-9059-9f09b6cdb4e5.png)
https://forum.theprodigy.ru/1388370

**Comment by rekcuFniarB** [comment]
Бля лол, новые блять, говнокодеры фейсбука отдают пользователям кучу ненужного технического мусора в составе ссылок, который где-то может не пройти парсинг, но виноваты не они конечно, гг. Предлагаю зарепортить им. Для ссылки достаточно этой части: https://www.facebook.com/rgorin.zx/posts/3303345403026764

**Comment by rekcuFniarB** [comment]
Алсо специально для сложных ссылок есть тег `[url]`, в который и следует оборачивать ссылки.

**Comment by DJ-ZX** [comment]
Речь не только проэто, там почему то символы в ссылке ломаются, потому и не парсится, не согласен с вердиктом

**Comment by DJ-ZX** [comment]
в гитхабе не ломается, например

**Comment by rekcuFniarB** [comment]
> в гитхабе не ломается, например

Потому что гитхаб не поддерживает 100500 всякой хрени, что поддерживается у нас, которая добавлялась в разное время разными людьми и может друг с другом где-то конфликтовать. У Гитхаба, к тому же, очень простая разметка markdown, в отличие от навороченного BB code.

Могу предложить удалить лишний функционал, это сейчас модно.

**Comment by rekcuFniarB** [comment]
Посмотрел на ссылку внимательней:
это и не баг, собственно. У тебя по сути BB код посреди ссылки, соответственно парсер это должен воспринимать как конец ссылки и дальше интерпретировать встретившийс BB код, что и происходит с твоей ссылкой. То есть парсер отработал корректно.

В данном случае следует оборачивать ссылку в тег `url`, который явно указыает где ссылка начинается и где заканчивается, чтобы парсер не пытался угадывать что за дичь ты пытаешься запостить.

**Comment by DJ-ZX** [comment]
не так - считаю, что надо считать ссылкой весь плейн текст без пробелов после хттпс по умолчанию
то есть вот так должен работать дефолтный парсер - хттпс://ааааааааааааааааааааа..ааааа.ааааааааазхщвпывщъзыаааааааа - это должно быть ссылкой автоматом, после проверки или остальные правила не сработали

**Comment by rekcuFniarB** [comment]
> считаю, что надо считать ссылкой весь плейн текст без пробелов после хттпс по умолчанию

Ага и результат парсинга будет такой: [https://example.com/qwerty[b]крым наш[/b]](https://example.com/qwerty[b]крым%20наш[/b])

Я хз зачем ты переоткрывашь тикет, почему так работает и почему для парсера это корректное поведение  я объяснил. И тем более не вижу смысла затачивать движок под тупой контингент, для которых и так есть ВК, Фейсбук и т.п., где нет продвинутых возможностей по разметке сообщений. Это так же тупо как требовать чтобы MS Word был простым как Блокнот.

Можно конечно придумать опцию в настройках профиля "режим для тупых" / "режим для продвинутых", но я не вижу смысла тратить на это время только потому что тебе не хватает ума заключить ссылку как полагается в тег `[url][/url]`.

**Comment by rekcuFniarB** [comment]
> считаю, что надо считать ссылкой весь плейн текст без пробелов после хттпс по умолчанию

Считает он... Я тут ещё вспомнил что ты недавно наоборот утверждал что надо чтобы ссылки не парсились дальше запятых, круглых скобок и ещё каких-то символов, потому что у тупых не хватает ума правильно воспользоваться разметкой и у них случайно ссылка захватывает лишние символы, например [https://example.com/qwerty,](https://example.com/qwerty,) ([https://example.com/)](https://example.com/%29) и т.п.

Тогда я тоже отказался что-то менять, т.к. запятые и круглые скобки в составе урлов это норма и они встречаются, наример в ссылках википедии https://ru.wikipedia.org/wiki/%D0%A0%D0%B0%D1%88%D0%BA%D0%B0_(%D0%B3%D0%BE%D1%81%D1%83%D0%B4%D0%B0%D1%80%D1%81%D1%82%D0%B2%D0%BE)
А запятые, например, используются в URL'ах SMF: https://forum.thechembase.com/index.php/topic,60.msg23495.html

**Comment by DJ-ZX** [comment]
да, надо как на ресурсах для тупых, я считаю, собственно баг и про это

**Comment by rekcuFniarB** [comment]
Не надо. Иначе нах этот форум вообще нужен, если можно и так торчать в ВК, Фейсбуке, Ок.ру и т.п. Пойду лучше с Биллом Гейтсом ра Реддите торчать.
